packages/config/src/api/site_info.js

```diff
@@ -15,26 +18,31 @@ export const getSiteInfo = async function ({
   siteId,
   mode,
   siteFeatureFlagPrefix,
+  featureFlags = {},
   testOpts: { env: testEnv = true } = {},
+  logs,
 }) {
   if (api === undefined || mode === 'buildbot' || !testEnv) {
     const siteInfo = siteId === undefined ? {} : { id: siteId }
     return { siteInfo, accounts: [], addons: [] }
   }
+  const fetchIntegrations = featureFlags.buildbot_fetch_integrations
+
+  const promises = [getSite(api, siteId, siteFeatureFlagPrefix), getAccounts(api), getAddons(api, siteId)]
+
+  if (fetchIntegrations) {
+    promises.push(getIntegrations({ api, owner: 'site', ownerId: siteId, logs }))
+  }
 
-  const [siteInfo, accounts, addons] = await Promise.all([
-    getSite(api, siteId, siteFeatureFlagPrefix),
-    getAccounts(api),
-    getAddons(api, siteId),
-  ])
+  const [siteInfo, accounts, addons, integrations] = await Promise.all(promises)
```

Comment:
mega nit: I think it would be nicer to handle the default value here and not at the end of the function, because it ensures the variable has a consistent value from the moment it's created and it keeps the `return` statement just about returning the stuff that's already been computed. 

```suggestion
  const [siteInfo, accounts, addons, integrations = []] = await Promise.all(promises)
```

Reply:
so much nicer, thanks